src/hotspot/share/utilities/ostream.hpp

```diff
@@ -42,7 +42,7 @@ DEBUG_ONLY(class ResourceMark;)
 //     jio_fprintf(defaultStream::output_stream(), "Message");
 // This allows for redirection via -XX:+DisplayVMOutputToStdout and
 // -XX:+DisplayVMOutputToStderr
-class outputStream : public ResourceObj {
+class outputStream : public CHeapObj<mtInternal> {
```

Comment:
Were you going to change this to mtInvalid to require all components to pass the appropriate mtComponent flag?  People might bikeshed on the name mtInvalid - it could also be mtOverride or mtDynamic.  In any case, this is a nice improvement.